crates/noirc_evaluator/src/brillig/mod.rs

```diff
@@ -46,15 +49,46 @@ impl std::ops::Index<FunctionId> for Brillig {
 }
 
 impl Ssa {
-    /// Generate compilation artifacts for brillig functions
+    /// Compile to brillig brillig functions and ACIR functions reachable from them
     pub(crate) fn to_brillig(&self) -> Brillig {
-        // Collect all of the brillig functions
-        let brillig_functions =
-            self.functions.values().filter(|func| func.runtime() == RuntimeType::Brillig);
+        // Collect all the function ids that are reachable from brillig
+        // That means all the functions marked as brillig and ACIR functions called by them
+        let mut reachable_function_ids: HashSet<FunctionId> = HashSet::new();
+
+        // Initialize the queue with all the functions marked as brillig
+        let mut reachability_queue: Vec<FunctionId> = self
```

Comment:
What are your thoughts on filter.map? Something like:

```rust
let mut reachability_queue: Vec<FunctionId> = self
    .functions
    .iter()
    .filter(|(_, func)| func.runtime() == RuntimeType::Brillig)
    .map(|(id, _)| *id)
    .collect();
```

Reply:
Don't mind either way, but I recall doing filter().map() instead of filter_map and clippy complaning about it in one of our repos https://rust-lang.github.io/rust-clippy/master/index.html#/manual_filter_map

Reply:
Ah I'm suprised it would complain for the above; if so, then lets leave as is!

Reply:
Oh, it complained because i was using filter_map with options, instead of checking an arbitrary condition like here. Whatever you think is more readable!